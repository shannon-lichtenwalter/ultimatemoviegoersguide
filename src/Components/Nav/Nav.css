nav {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  position: fixed;
  top: 0;
  width: 100%;
  height: 150px;
  background-image: url('https://images.unsplash.com/photo-1558544109-58992f348795?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=848&q=80');
  background-position: bottom;
  background-repeat: no-repeat;
  background-size: cover;
  z-index: 10;
    -o-box-shadow:      12px 12px 20px #555;
    -icab-box-shadow:   12px 12px 20px #555;
    -khtml-box-shadow:  12px 12px 20px #555;
    -moz-box-shadow:    12px 12px 20px #555;
    -webkit-box-shadow: 12px 12px 20px #555;
    box-shadow:         12px 12px 20px #555;
}

nav > button {
  margin: 10px;
  border: none;
  border-radius: 3px;
  background-color: transparent;
  font-size: 1rem;
}

nav > button:hover {
  border: 1px solid #102619;
}

@media screen and (min-width: 580px) {
  nav {
    justify-content: flex-start;
    align-items: flex-end;
  }
}