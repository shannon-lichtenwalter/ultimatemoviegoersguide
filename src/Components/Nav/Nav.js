import React from 'react';
import { withRouter } from 'react-router-dom';
import './Nav.css';

class Nav extends React.Component{
  render(){
    return(
      <nav>
        <button onClick={() => this.props.history.push('/')}>Home</button>
        <button onClick={() => this.props.history.push('/search')}>Search</button>
        <button onClick={() => this.props.history.push('/about')}>About</button>
      </nav>
    )
  }
}

export default withRouter(Nav);