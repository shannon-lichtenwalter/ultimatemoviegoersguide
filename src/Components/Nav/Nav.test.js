import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Nav from './Nav';

describe('Nav Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter><Nav /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the Nav component correctly when rendered', () => {
    const wrapper = mount(<BrowserRouter><Nav /></BrowserRouter>);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});