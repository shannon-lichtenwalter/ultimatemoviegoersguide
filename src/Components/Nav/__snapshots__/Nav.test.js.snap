// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component Displays the Nav component correctly when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Nav)>
      <Nav
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <nav>
          <button
            onClick={[Function]}
          >
            Home
          </button>
          <button
            onClick={[Function]}
          >
            Search
          </button>
          <button
            onClick={[Function]}
          >
            About
          </button>
        </nav>
      </Nav>
    </withRouter(Nav)>
  </Router>
</BrowserRouter>
`;
